L1Trigger/Phase2L1ParticleFlow/src/BJetId.cc

```
@@ -0,0 +1,95 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/BJetId.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include <cmath>
+
+static constexpr unsigned int n_particles_max = 10;
+
+BJetId::BJetId(const std::string &iInput, const BJetTFCache *cache, const std::string &iWeightFile, int iNParticles) {
+  NNvectorVar_.clear();
+  session_ = tensorflow::createSession(cache->graphDef.get());
+  fNParticles_ = iNParticles;
```

why do you need both a hard-coded `n_particles_max` and a configurable `fNParticles_`?
also, you may want to throw an exception if `iNParticles` is larger than `n_particles_max`.

in fact, I guess it should be possible to derive what is the correct max number of particles by looking at the dimension of the tensorflow graph, and that could replace the hard-coded `n_particles_max`.

---

I think this was just a holdover from the tau NN code I started from. I have just removed `n_particles_max` since it shouldn't really be necessary.

I took a quick look and it wasn't obvious how to extract the correct number from the tf graph, so I've left it to the user for now.